astropy/table/serialize.py

```diff
@@ -35,10 +37,13 @@ class SerializedColumn(dict):
 def _represent_mixin_as_column(col, name, new_cols, mixin_cols,
                                exclude_classes=()):
     """Convert a mixin column to a plain columns or a set of mixin columns."""
+    obj_attrs = col.info._represent_as_dict()
+    ordered_keys = col.info._represent_as_dict_attrs
+
     # If not a mixin, or if class in ``exclude_classes`` tuple then
     # treat as a normal column.  Excluded sub-classes must be explicitly
     # specified.
-    if not has_info_class(col, MixinInfo) or col.__class__ in exclude_classes:
```

Comment:
This is a key change here to not use "being a mixin" as the criteria for getting into the serialize code, instead ask the object if it has columns that need to be serialized separately.

Reply:
Ah, yes OOP to the rescue!

Reply:
Maybe adjust the comment above this line to reflect the change.